File: Metode-Avansate-de-Programare/lab2/decorator/StrategyTaskRunner.java
package sem1_2.decorator;

import sem1_2.factory.Strategy;
import sem1_2.model.Container;
import sem1_2.model.Task;
import sem1_2.factory.TaskContainerFactory;

public class StrategyTaskRunner implements TaskRunner {

    private Container container;

    public StrategyTaskRunner(Strategy strategy) {
        container = TaskContainerFactory.getInstance().createContainer(strategy);
    }

    @Override
    public void executeOneTask() {
        Task task = container.remove();
        task.execute();
    }

    @Override
    public void executeAll() {
        while (hasTask()) {
            executeOneTask();
        }
    }

    @Override
    public void addTask(Task t) {
        container.add(t);
    }

    @Override
    public boolean hasTask() {
        return !container.isEmpty();
    }
}

File: Metode-Avansate-de-Programare/lab11/lab11/StrategyTaskRunner.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace lab11;

public class StrategyTaskRunner : TaskRunner
{
    private Container container;

    public StrategyTaskRunner(Strategy strategy)
    {
        container = TaskContainerFactory.getInstance().createContainer(strategy);
    }

    public void executeOneTask()
    {
        Task task = container.remove();
        Console.WriteLine(task.TaskID);
        task.execute();
    }

    public void executeAll()
    {
        while (hasTask())
        {
            executeOneTask();
        }
    }

    public void addTask(Task t)
    {
        container.add(t);
    }

    public bool hasTask()
    {
        return !container.isEmpty();
    }
}